File: 682-baseball-game/682-baseball-game.py
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        history = []
        for o in ops:
            if o == 'C':
                history.pop()
            elif o == 'D':
                history.append(history[-1] * 2)
            elif o == '+':
                history.append(history[-1] + history[-2])
            else:
                history.append(int(o))
        return sum(history)
                

File: 682-baseball-game/682-baseball-game.java
class Solution {
    /*["-42","90","82","+","C","-21","43","D","D","-32","-98","83","D","-86","68","68","D","C","C","7"]
    */
    public int calPoints(String[] ops) {
        int res = 0;
        List<Integer> l = new ArrayList<>();
        for(String s : ops){
            if("+".equals(s)){
                l.add(l.get(l.size()-1) + l.get(l.size()-2));
                res += l.get(l.size() -1);
            }
            else if("C".equals(s)){
                res -= l.get(l.size() - 1);
                l.remove(l.size()-1);
            }
            else if("D".equals(s)){
                l.add(l.get(l.size()-1) * 2);
                res += l.get(l.size() - 1);
            }
            else{
                l.add(Integer.parseInt(s));
                res += l.get(l.size() - 1);
            }
        }
        return res;
    }
}